ni_measurementlink_service/measurement/service.py

```diff
@@ -377,12 +384,21 @@ def _make_annotations_dict(
 
         return annotations
 
-    def _enum_to_annotations_value(self, enum_type: Type[Enum]) -> str:
-        if not any(member.value == 0 for member in enum_type):
-            raise ValueError("The enum does not have a value for 0.")
+    def _enum_to_annotations_value(self, enum_type: SupportedEnumType) -> str:
         enum_values = {}
-        for member in enum_type:
-            enum_values[member.name] = member.value
+        # Note that the type of protobuf enums are an instance of EnumTypeWrapper
+        # in addition to being a type itself. Additionally, issubclass excludes
+        # instances as valid parameters so we use isinstance here.
+        if isinstance(enum_type, EnumTypeWrapper):
+            if not any(value.number == 0 for value in enum_type.DESCRIPTOR.values):
+                raise ValueError("The enum does not have a value for 0.")
+            for value in enum_type.DESCRIPTOR.values:
+                enum_values[value.name] = value.number
```

Comment:
`EnumTypeWrapper` provides `dict`-style methods that are more concise than accessing the `EnumDescriptor` directly.

```
            if 0 not in enum_type.values():
                raise ValueError("The enum does not have a value for 0.")
            for name, value in enum_type.items():
                enum_values[name] = value
```

Reply:
Refactored as suggested. Note that I also had to add a cast here to resolve mypy issue complaining about `Union[Type[Enum], EnumTypeWrapper]` not having `values` or `items` because we swapped the `isinstance` check to not reference `EnumTypeWrapper` directly anymore.

Reply:
You can remove the `if TYPE_CHECKING` around the cast by passing the type name as a string :)

![image](https://github.com/ni/measurementlink-python/assets/17914399/c74964cd-d33a-4f94-9aa6-f8d27371b20e)

Typeshed confirms that `cast` has an overload that takes `typ: str`.
https://github.com/python/typeshed/blob/main/stdlib/typing.pyi#L825

Reply:
Swapped to casting by name rather than type as suggested.